# (1) Pedestrian Dataset Cleaning
Our overall plan in the data team, is to divide up the borough of manhattan into taxi zones and then from there, predict the busyness of each zone based on known factors. Therefore, the pedestrian trip data may enhance part of our investigations. 


In [1]:
import numpy as np
import pandas as pd
import pyarrow as pa
import pyarrow.parquet as pq
import matplotlib.pyplot as plt
from IPython.display import display

C:\Users\natha\anaconda3\lib\site-packages\pandas\core\computation\expressions.py:21: UserWarning: Pandas requires version '2.8.4' or newer of 'numexpr' (version '2.8.3' currently installed).
  from pandas.core.computation.check import NUMEXPR_INSTALLED
C:\Users\natha\anaconda3\lib\site-packages\pandas\core\arrays\masked.py:60: UserWarning: Pandas requires version '1.3.6' or newer of 'bottleneck' (version '1.3.5' currently installed).
  from pandas.core import (


In [26]:
# read in data from csv file to pandas dataframe. 
df = pd.read_csv("PedCountLocations.csv",  keep_default_na=True, delimiter=',', skipinitialspace=True)

In [27]:
# Check number of rows and columns
num_rows, num_cols = df.shape
print(f"Number of rows: {num_rows}, Number of columns: {num_cols}")

Number of rows: 114, Number of columns: 104


In [28]:
# Print the first and last 5 rows
print("\nFirst 5 rows:")
df.head()


First 5 rows:


the_geom  OBJECTID  Location Borough  \
0   POINT (-73.90459140730678 40.87919896648575)         1         1   Bronx   
1   POINT (-73.92188432870218 40.82662794123293)         2         2   Bronx   
2   POINT (-73.89535781584335 40.86215460031517)         3         3   Bronx   
3    POINT (-73.87892467324478 40.8812869959873)         4         4   Bronx   
4  POINT (-73.88956389732787 40.844636776717685)         5         5   Bronx   

            Street_Nam         From_Stree          To_Street Iex  May07_AM  \
0             Broadway  West 231st Street     Naples Terrace   N    1189.0   
1    East 161st Street      Gra Concourse    Sheridan Avenue   Y    1511.0   
2    East Fordham Road   Valentine Avenue     Tiebout Avenue   Y    1832.0   
3   East Gun Hill Road  Bainbridge Avenue  Rochambeau Avenue   N     764.0   
4  East Tremont Avenue    Prospect Avenue     Clinton Avenue   N     650.0   

   May07_PM  ...  May22_MD  Oct22_AM  Oct22_PM  Oct22_MD  May23_AM  May23_pM  \
0    4094.0  ...    1883.0     745.0    2637.0    2241.0    1897.0    5018.0   
1    3184.0  ...    2168.0    2608.0    4277.0    1867.0    2384.0    4600.0   
2   12311.0  ...    4987.0    1662.0    6572.0    4765.0    1654.0    6918.0   
3    2673.0  ...    1063.0    1448.0    2120.0    1032.0    1398.0    2280.0   
4    2538.0  ...    1606.0     774.0    2084.0    1611.0     730.0    2077.0   

   May23_MD  Oct23_AM  Oct23_PM  Oct23_MD  
0    2496.0     935.0    3759.0    1745.0  
1    2064.0    2822.0    5050.0    1287.0  
2    4696.0    1560.0    6880.0    2453.0  
3    1303.0    1404.0    2089.0     807.0  
4    1578.0     786.0    2140.0    1115.0  

[5 rows x 104 columns]

# Format of CSV

We can see that the 114 rows are indicating 114 different locations and the rows are 104 columns in length. We should reformat this to match our taxi dataset. 